### PR TITLE
[exporter/coralogixexporter] Add OpenTelemetry based trace client and deprecate the old jaeger based client.

### PR DESCRIPTION
**Pull request by povilasv**
**Description:** <Describe what has changed.>

Update trace client to use OpenTelemetry

**Link to tracking Issue:** #7931 
**Testing:** <Describe what testing was performed and which tests were added.>

- unit tests
- manual tests

**Documentation:** <Describe the documentation added.>

Updated docs

### CODE REVIEW

File: exporter/coralogixexporter/traces_client.go
Comment (frzifus) [inline exporter/coralogixexporter/traces_client.go:47]:
```suggestion
	oCfg, ok := cfg.(*Config)
	if !ok {
		return nil, fmt.Errorf("invalid config exporter, expect type: %T, got: %T". &Config{}, cfg)
	}
```
mh.. We could check the type here to avoid some trouble in the future. What do you think?

File: exporter/coralogixexporter/traces_client.go
Comment (frzifus) [inline exporter/coralogixexporter/traces_client.go:65]: Is that a debug leftover?

File: exporter/coralogixexporter/traces_client.go
Comment (frzifus) [inline exporter/coralogixexporter/traces_client.go:58]:
```suggestion
func (e *tracesExporter) start(_ context.Context, host component.Host) error {
```
Is there a reason to use a named return statement here? Maybe we should avoid it to be consistent with other impl. parts.

**Review by frzifus — COMMENTED** [review]
thx @povilasv! I had already a brief look and let some comments.

**Review by frzifus — APPROVED** [review]
n1, up to the others 

cc @jpkrohling

**Review by oded-dd — APPROVED** [review]

File: exporter/coralogixexporter/README.md
Comment (codeboten) [inline exporter/coralogixexporter/README.md:44]: This is specifically the newer versions of the coralogix exporter correct?

File: exporter/coralogixexporter/README.md
Comment (codeboten) [inline exporter/coralogixexporter/README.md:46]:
```suggestion
To migrate, please remove the old endpoint field, and change the configuration to `traces.endpoint` using the new Tracing endpoint.
```

File: exporter/coralogixexporter/README.md
Comment (codeboten) [inline exporter/coralogixexporter/README.md:58]:
```suggestion
exporters
  coralogix:
```

**Review by codeboten — COMMENTED** [review]
Just a few comments, please rebase the PR.

File: exporter/coralogixexporter/config.go
Comment (jpkrohling) [inline exporter/coralogixexporter/config.go:40]: Isn't this bringing a breaking change to your users? You should follow the deprecation guidelines instead of just breaking your user's configuration.

**Review by jpkrohling — CHANGES_REQUESTED** [review]
This should follow the breaking changes guideline.

File: exporter/coralogixexporter/config.go
Comment (jpkrohling) [reply exporter/coralogixexporter/config.go:40]: Breaking changes are allowed, but we have a deprecation policy: https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#breaking-changes

I'm not comfortable approving this PR with known breaking changes ignoring our policy. If other approvers are OK with that, I won't block this.

cc @open-telemetry/collector-contrib-approvers

Comment (oded-dd) [reply exporter/coralogixexporter/config.go:40]: @jpkrohling You are correct, no argue here, but it is based on a request that came from all of our customers using the endpoint, which was communicated and agreed this breaking change is not "breaking change" for their use case, as it is currently only acts as jaeger endpoint. All of them are waiting for the PR to be merged and released. 
I agree that from a technical perspective it should be changed according to the deprecated policy, but from a business perspective it should be enough to add what @povilasv already added in the docs. We can argue the importance of a business perspective when it comes to opensource, but in the end it should have some weight. 

Anyways, we will change it accordingly, but it is important for future reference.

Comment (jpkrohling) [reply exporter/coralogixexporter/config.go:40]: Keeping it backward compatible is really, really easy in this case. The effort to discuss this was already higher than implementing the change itself, I'm afraid ;-) Just add the new structure, and check during the bootstrap whether the old one is being used or the new one (or both). If only the old one is used, just copy it over to the new one, logging a warning message asking users to start using the new config property. The remaining of your code can use the new location already. You'd then keep both audiences happy.

File: exporter/coralogixexporter/go.mod
Comment (jpkrohling) [inline exporter/coralogixexporter/go.mod:12]: You might want to revert this.

File: unreleased/coralogix-trace.yaml
Comment (jpkrohling) [inline unreleased/coralogix-trace.yaml:16]:
```suggestion
subtext: "Please remove the `endpoint` field and use the new `traces.endpoint` field with your OpenTelemetry endpoint."
```

**Review by jpkrohling — COMMENTED** [review]
Other than the go.mod changes that need to be reverted, there's a small nit on the changelog entry. LGTM otherwise.

I didn't manually check the backward compatibility, I'm trusting the current test suite is ensuring the current clients will still work with the new property.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
